text/0000-relax-const-restrictions.md

```diff
@@ -0,0 +1,74 @@
+- Feature Name: relax_const_restrictions
+- Start Date: 2022-10-06
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Allow `const` functions to behave differently during constant-evaluation and runtime and remove all restrictions from `const` functions if they are called at runtime.
+
+# Motivation
+[motivation]: #motivation
+
+The past restriction of `const` functions having to behave the same way no matter where they were called has been a limitation and soundness issue. While the Rust language does, at the time of writing, not expose a way to determine whether a function has been called during constant evaluation or runtime (and this RFC does not propose adding such a feature), such an intrinsic (`const_eval_select`) does currently exist internally in `rustc`.
```

Comment:
"soundness issue" is very general. Maybe "has been unclear whether such a difference in behaviour could cause unsoundness".